src/TwitchLib.Communication/Clients/ClientBase.cs

```diff
@@ -37,7 +37,7 @@ public abstract class ClientBase<T> : IClient
         
         internal static TimeSpan TimeOutEstablishConnection => TimeSpan.FromSeconds(15);
 
-        protected ILogger? Logger { get; }
+        protected readonly ILogger? Logger;
```

Comment:
This should be basically the same thing. Is there any performance gain in this?

Reply:
I don't think so, (maybe very very small). The reason why I changed it is for future compatibility (you can define logging methods directly in the class\*, but right now, you will have an error if the ILogger is a property and not a field - this looks like bug from MS side).

\* there is another problem, you cannot use "logging SourceGen" when the logger is set to null, but we can simply set ILogger to `NullLoger` and suppress all warnings with `!`